lib/if.c

```diff
@@ -668,21 +674,26 @@ int if_is_running(const struct interface *ifp)
    if ptm checking is enabled, then ptm check has passed */
 int if_is_operative(const struct interface *ifp)
 {
-	return ((ifp->flags & IFF_UP)
-		&& (((ifp->flags & IFF_RUNNING)
-		     && (ifp->ptm_status || !ifp->ptm_enable))
-		    || !CHECK_FLAG(ifp->status,
-				   ZEBRA_INTERFACE_LINKDETECTION)));
+	return ((ifp->flags & IFF_UP) &&
+		(((ifp->flags & IFF_RUNNING)
+#ifdef IFF_LOWER_UP
+		  && (ifp->flags & IFF_LOWER_UP)
```

Comment:
1. this will break FRR completely on older kernels that do not send this flag (you are making a compile-time check, that does not mean the [different] kernel at run-time will support the flag)

2. AFAIK `IFF_RUNNING` should never be set if `IFF_LOWER_UP` is not set? Can you elaborate when this is not the case?

Reply:
I've crossed out the compatibility part, since the flag has been in the kernel since ≤2014. The other question remains.

Reply:
> 2\. AFAIK `IFF_RUNNING` should never be set if `IFF_LOWER_UP` is not set? Can you elaborate when this is not the case?

According to the kernel documentation:  https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/networking/operstates.rst?h=v6.7-rc8#n29

```
ifinfomsg::if_flags & IFF_RUNNING:
 Interface is in RFC2863 operational state UP or UNKNOWN. [...]
ifinfomsg::if_flags & IFF_LOWER_UP:
 Driver has signaled netif_carrier_on()
```
Therefore, it is consistent with the kernel specifications for an interface to have `IFF_RUNNING` set without having `IFF_LOWER_UP` set.